aten/src/ATen/core/VariableFallbackKernel.cpp

```diff
@@ -0,0 +1,40 @@
+#include <ATen/core/dispatch/Dispatcher.h>
+#include <ATen/core/LegacyTypeDispatch.h>
+
+/*
+ * This file implements a variable fallback kernel for custom operators.
+ * Since tensors always have the VariableTensorId set, but custom operators
+ * usually don't have a kernel registered for VariableTensorId, the dispatcher
+ * will call into this fallback kernel instead.
+ * Note that this is not a correct autograd implementation. It will just
+ * fallthrough to the custom operator implementation.
+ * If you want a custom operator to work with autograd, you need to use
+ * autograd::Function so that the custom operator implementation knows how to
+ * do autograd.
+ * Note also that ops from native_functions.yaml register their own variable
+ * kernels, so this is never called for them.
+ */
+
+// TODO This whole file should be deleted and replaced with the mechanism
+//      described in https://github.com/pytorch/pytorch/issues/29548
+
+using c10::OperatorHandle;
+using c10::Stack;
+using c10::TensorTypeId;
+using c10::TensorTypeSet;
+using c10::Dispatcher;
+using c10::KernelFunction;
+
+namespace {
+
+void variable_fallback_kernel(const OperatorHandle& op, Stack* stack) {
+    at::AutoNonVariableTypeMode _var_guard(true);
+    Dispatcher::singleton().callBoxed(op, stack);
```

Comment:
OK, this isn't really what I've been advocating for, but I'll let it slide for now.

Reply:
it's not? IIRC, we said we're setting the thread local flag and re-dispatching. That's what I'm doing here. What would you do instead?

Reply:
We also need to setup `grad_fn` on the outputs to error if you try to backprop through them! Otherwise you'll just poop out `requires_grad=False` outputs which will silently fail to backprop if someone uses them.

Reply:
Ah, that's what you mean. Yes, we should do that. This doesn't regress behavior though, it's just as broken as it was before. There were some issues with adding the grad_fn if I remember correctly which meant we can't do it right now, but I'll look at it in a separate PR.

Reply:
Yep, that's why I approved.